src/engraving/rendering/score/tupletlayout.cpp

```diff
@@ -45,124 +45,232 @@ void TupletLayout::layout(Tuplet* item, LayoutContext& ctx)
+        }
+    }
+    // layout t
+    layout(t, ctx);
+}
+
+bool TupletLayout::isTopTuplet(ChordRest* cr)
+{
+    Tuplet* t = cr->tuplet();
+    if (t && t->elements().front() == cr) {
+        // find top level tuplet
+        while (t->tuplet()) {
+            t = t->tuplet();
+        }
+        // consider tuplet cross if anything moved within it
+        if (t->cross()) {
+            return false;
         } else {
-            item->number()->setXmlText(numberString);
+            return true;
         }
+    }
 
-        item->setIsSmall(true);
-        for (const DurationElement* e : item->elements()) {
-            if ((e->isChordRest() && !toChordRest(e)->isSmall()) || (e->isTuplet() && !toTuplet(e)->isSmall())) {
-                item->setIsSmall(false);
-                break;
-            }
+    // no tuplet or not first element
+    return false;
+}
+
+bool TupletLayout::notTopTuplet(ChordRest* cr)
+{
+    Tuplet* t = cr->tuplet();
+    if (t && t->elements().front() == cr) {
+        // find top level tuplet
+        while (t->tuplet()) {
+            t = t->tuplet();
         }
-        item->number()->mutldata()->setMag(item->isSmall() ? style.styleD(Sid::smallNoteMag) : 1.0);
-    } else {
+        // consider tuplet cross if anything moved within it
+        if (t->cross()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // no tuplet or not first element
+    return false;
+}
+
+void TupletLayout::createNumber(Tuplet* item, LayoutContext& ctx)
+{
+    if (item->numberType() == TupletNumberType::NO_TEXT) {
         if (item->number()) {
             if (item->number()->selected()) {
                 ctx.deselect(item->number());
             }
             delete item->number();
             item->setNumber(nullptr);
         }
+
+        return;
     }
-    //
-    // find out main direction
-    //
-    if (item->direction() == DirectionV::AUTO) {
-        int up = 0;
-        for (const DurationElement* e : item->elements()) {
-            if (e->isChord()) {
-                const Chord* c = toChord(e);
-                if (c->stemDirection() != DirectionV::AUTO) {
-                    up += c->stemDirection() == DirectionV::UP ? 1000 : -1000;
-                } else {
-                    up += c->up() ? 1 : -1;
-                }
-            }
+
+    if (!item->number()) {
+        Text* number = Factory::createText(item, TextStyleType::TUPLET);
+        number->setComposition(true);
+        number->setTrack(item->track());
+        number->setParent(item);
+        number->setVisible(item->visible());
+        number->setColor(item->color());
+        item->setNumber(number);
+        item->resetNumberProperty();
+    }
+
+    // tuplet properties are propagated to number automatically by setProperty()
+    // but we need to make sure flags are as well
+    item->number()->setPropertyFlags(Pid::FONT_FACE, item->propertyFlags(Pid::FONT_FACE));
```

Comment:
I know this isn't new, but I don't like it! I think it would be worth us taking a look at property delegates and moving some logic out of `Spanner` and `SpannerSegment` to `EngravingItem`. 
Probably not a job for this PR though